src/highlight.c

```diff
@@ -309,8 +309,8 @@ static char *(highlight_init_dark[]) = {
     CENT("SignColumn term=standout ctermbg=DarkGrey ctermfg=Cyan",
 	 "SignColumn term=standout ctermbg=DarkGrey ctermfg=Cyan guibg=Grey guifg=Cyan"),
 #endif
-    CENT("Visual term=reverse",
-	 "Visual term=reverse guibg=DarkGrey"),
+    CENT("Visual ctermbg=Grey ctermfg=Black",
+	 "Visual ctermbg=Grey ctermfg=Black guibg=DarkGrey"),
```

Comment:
As you mentioned in `bg=dark` gui visual looks not that great, we can make it `grey34`:

![image](https://github.com/vim/vim/assets/234774/24fbcfb8-714c-4ec7-bd53-15bc3530a3b6)